src/agent/rustjail/src/container.rs

```diff
@@ -624,9 +624,9 @@ fn do_init_child(cwfd: RawFd) -> Result<()> {
     // notify parent that the child's ready to start
     write_sync(cwfd, SYNC_SUCCESS, "")?;
     log_child!(cfd_log, "ready to run exec");
-    unistd::close(cfd_log);
-    unistd::close(crfd);
-    unistd::close(cwfd);
+    unistd::close(cfd_log)?;
+    unistd::close(crfd)?;
```

Comment:
It's better not change the logic in a warning fix commit
what about use `unistd::close(cfd_log).ok()` or `let _ = unistd::close(cfd_log)` instead

Reply:
> It's better not change the logic in a warning fix commit

I strongly disagree, if that means suppressing the warning without actually fixing the problem its pointing out.

Reply:
@dgibson Fix the problem its pointing out in a PR named by the problem not in this general PR named "Remove compilation warnings"

Reply:
@c3d Why do you think it is necessary to break the function and report error.

Reply:
@Tim-Zhang Your argument about not changing the logic makes sense. I'm revisiting these changes, will open separate issues for them.